src/ResourceManager/KeyVault/Commands.KeyVault/Commands/VirtualNetworkRuleSet/AddAzureKeyVaultNetworkRuleSet.cs

```diff
@@ -0,0 +1,157 @@
+// ----------------------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.KeyVault.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    /// <summary>
+    /// Add network rule set
+    /// NOTE: Define VaultName & ResourceGroupName in this class instead of base one because TAB order for input.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Add, "AzureRmKeyVaultNetworkRuleSet",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = ByVaultNameParameterSet)]
+    [OutputType(typeof(PSKeyVault))]
+    public class AddAzureKeyVaultNetworkRuleSet : KeyVaultNetworkRuleSetBase
+    {
+        private const string ByVaultNameParameterSet = "ByVaultName";
+        private const string ByInputObjectParameterSet = "ByInputObject";
+        private const string ByResourceIdParameterSet = "ByResourceId";
+
+        #region Input Parameter Definitions
+        /// <summary>
+        /// Vault name
+        /// </summary>
+        [Parameter(Mandatory = true,
+            Position = 0,
+            HelpMessage = "Specifies the name of a key vault whose network rule is being modified.")]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        /// <summary>
+        /// KeyVault object
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 0,
+                   ParameterSetName = ByInputObjectParameterSet,
+                   ValueFromPipeline = true,
+                   HelpMessage = "KeyVault object")]
+        [ValidateNotNullOrEmpty]
+        public PSKeyVault InputObject { get; set; }
+
+        /// <summary>
+        /// KeyVault ResourceId
+        /// </summary>
+        [Parameter(Mandatory = true,
+                   Position = 0,
+                   ParameterSetName = ByResourceIdParameterSet,
+                   ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "KeyVault Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        /// <summary>
+        /// Resource group name
+        /// </summary>
+        [Parameter(Mandatory = false,
+            Position = 1,
+            ParameterSetName = ByVaultNameParameterSet,
+            HelpMessage = "Specifies the name of the resource group associated with the key vault whose network rule is being modified.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public string ResourceGroupName { get; set; }
+        #endregion
+
+        public override void ExecuteCmdlet()
+        {
+            if (InputObject != null)
+            {
+                VaultName = InputObject.VaultName;
```

Comment:
You should be able to get the resourcegroupname both here and for ResourceId - that will be more efficient than querying the server below.